app/src/main/java/at/bitfire/icsdroid/ui/dialog/GenericAlertDialog.kt

```diff
@@ -0,0 +1,58 @@
+package at.bitfire.icsdroid.ui.dialog
+
+import androidx.compose.material.AlertDialog
+import androidx.compose.material.Text
+import androidx.compose.material.TextButton
+import androidx.compose.runtime.Composable
+
+/**
+ * Provided by [GenericAlertDialog], gives access to some interactions from inside the composables
+ * and callbacks.
+ */
+interface AlertDialogContext {
```

Comment:
Really cool, that this is possible and good to know. But I feel the complexity it adds in comparison do the benefit it brings is too small. I would keep this for more complicated compose setups and stick with the simple tools for now.